### ISSUE TITLE
[Travis-CI] compilation error "is private within this context boost::fusion::for_each"

### ISSUE THREAD

**Issue by tt4g** [tt4g/conan-soci#5]
```
/home/conan/.conan/data/SOCI/4.0.0/user/channel/build/aefaab8b4d1031a2ab4f98acd267dcfc1288fa89/soci/include/soci/bind-values.h:92:32: error: ‘soci::details::use_type_vector::use_sequence<T, void>::use_sequence(const soci::details::use_type_vector::use_sequence<T, void>&) [with T = boost::tuples::tuple<double, boost::optional<int>, std::basic_string<char> >]’ is private within this context
         boost::fusion::for_each(uc.t, use_sequence<T, details::no_indicator>(*this));
         ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by tt4g** [comment]
```
c:\users\appveyor\.conan\data\soci\4.0.0\user\channel\build\f10142afb17287698f280ed0f150dd08214c038e\soci\include\soci\bind-values.h(182): error C2280: 'soci::details::into_type_vector::into_sequence<T,soci::details::no_indicator>::into_sequence(const soci::details::into_type_vector::into_sequence<T,soci::details::no_indicator> &)': attempting to reference a deleted function [C:\Users\appveyor\.conan\data\SOCI\4.0.0\user\channel\build\f10142afb17287698f280ed0f150dd08214c038e\tests\postgresql\soci_postgresql_test.vcxproj]
                 with
                 [
                     T=boost::fusion::vector<double,boost::optional<int>,std::string>
                 ]
         c:\users\appveyor\.conan\data\soci\4.0.0\user\channel\build\f10142afb17287698f280ed0f150dd08214c038e\soci\include\soci\bind-values.h(170): note: see declaration of 'soci::details::into_type_vector::into_sequence<T,soci::details::no_indicator>::into_sequence'
                 with
                 [
                     T=boost::fusion::vector<double,boost::optional<int>,std::string>
                 ]
         c:\users\appveyor\.conan\data\soci\4.0.0\user\channel\build\f10142afb17287698f280ed0f150dd08214c038e\soci\include\soci\bind-values.h(170): note: 'soci::details::into_type_vector::into_sequence<T,soci::details::no_indicator>::into_sequence(const soci::details::into_type_vector::into_sequence<T,soci::details::no_indicator> &)': function was explicitly deleted
                 with
                 [
                     T=boost::fusion::vector<double,boost::optional<int>,std::string>
                 ]
```

**Comment by tt4g** [comment]
Related SOCI/soci/issues/679

**Comment by tt4g** [comment]
Temporary fix 993222e7f9141929c9dc19be87bd69d25ec4ffc8

**Comment by tt4g** [comment]
It has been fixed by SOCI.

Remove temporary fix 94f921870a0c8e5f4c137887dc94ead922f8d72d
